src/v/archival/ntp_archiver_service.cc

```diff
@@ -577,10 +632,11 @@ ss::future<ntp_archiver::scheduled_upload> ntp_archiver::schedule_single_upload(
         .delta_offset = delta,
         .ntp_revision = _rev,
         .archiver_term = _start_term,
+        .sname_format = cloud_storage::segment_name_format::v2,
       },
       .name = upload.exposed_name, .delta = offset - base,
       .stop = ss::stop_iteration::no,
-      .segment_read_lock = co_await upload.source->read_lock(segment_lock_deadline),
+      .segment_read_locks = co_await get_read_locks(upload, _segment_upload_timeout),
```

Comment:
changed so that now we have locks alongside the upload candidate, and then they are moved into the scheduled upload. so we have locks created as early as when we create upload candidate.